frameworks/shared/requirements.txt

```diff
@@ -4,8 +4,12 @@
 #
 #    pip-compile frameworks/shared/requirements.in
 #
+numpy==1.21.0
+    # via pyarrow
```

Comment:
I hope this is not introducing new version conflicts, the main advantage of `pickle` is that it was avoiding this : frameworks are supposed to install their dependencies after this one so they can always install another `numpy` version, hopefully still compatible with `pyarrow`… and in any case they can switch to alternatives (pickle, hdf, and json in the worst case)

Reply:
Since `pyarrow` only interfaces with `numpy` through its public functions (I should hope), it might be even _more_ compatible with various `numpy` versions?